pkg/credentialprovider/azure/azure_credentials.go

```diff
@@ -301,3 +302,26 @@ func (a *acrProvider) parseACRLoginServerFromImage(image string) string {
 
 	return ""
 }
+
+// shouldUseCache decide whether to use cache according to image and cache content
+func shouldUseCache(image string, d credentialprovider.DockerConfig) bool {
+	if len(d) > 2 {
+		// for service principal(len=5), always use cache
+		return true
+	}
+
+	fields := strings.Split(image, "/")
+	if len(fields) <= 1 || !strings.Contains(fields[0], azureCRName) {
```

Comment:
presence of a `.azurecr.` segment somewhere in the image doesn't seem correct... that would mean `bogus.azurecr.example.com` would match

Reply:
I think that's ok, that would waste one cache in every pull image operation.
since there is some secret azure ACR registry site would also contain *.azurecr.*, so the azurecr suffix endpoint could not be listed here, use a generic *.azurecr.* is a safe way to make sure it won't miss those secret registry sites